### ISSUE TITLE
multiple controllers setup randomly crashes when wifi connects

### ISSUE THREAD

**Issue by sebdelsol** [FastLED/FastLED#1153]
On my ESP32 with WS2812B strips, FastLED 3.4.0 _randomly_ crashes in [ESP32RMTController::startNext](https://github.com/FastLED/FastLED/blob/2fe6aebc778a836a8556b8e4508381e9c847f7a3/src/platforms/esp/32/clockless_rmt_esp32.cpp#L200) when Wifi connects.

It works fine :
- With FastLED 3.3.3,
- OR with a single FastLED controller,
- OR when Wifi is connected before any `FastLED.show()` call,
- OR when Wifi settings are saved in RAM instead of NVS.

### Code
```c++
#include <FastLED.h>
#include <WiFi.h>

// uncomment ANY of those to prevent the crash from happening
// #define WIFI_CFG_SAVED_IN_RAM
// #define WIFI_CONNECT_BEFORE_LOOP
// #define FASTLED_SINGLE_CONTROLLER

// fill up your Wifi ssid & password 
#define SSID        "*********"
#define PASSWORD    "*********"

#define SERIAL_BAUD 115200

#define NUM_LEDS    1
#define DATA_PIN1   12
#define DATA_PIN2   13

CRGB leds[NUM_LEDS];

void setup()
{
  Serial.begin(SERIAL_BAUD);

  // DO NOT CRASH when Wifi settings are saved in RAM instead of NVS
  #ifdef WIFI_CFG_SAVED_IN_RAM
    WiFi.persistent(false);
    Serial.println("Do not Crash: Wifi settings are saved in RAM instead of NVS");
  #endif

  // DO NOT CRASH when Wifi is connected before the loop 
  auto wifiCb = [] (WiFiEvent_t e, WiFiEventInfo_t i) { Serial.println("Wifi Connected, CRASH avoided, play again"); }; 
  WiFi.onEvent(wifiCb, SYSTEM_EVENT_STA_CONNECTED);
  WiFi.begin(SSID, PASSWORD); 

  #ifdef WIFI_CONNECT_BEFORE_LOOP
    while(WiFi.status() != WL_CONNECTED) { delay(100); };
    Serial.println("Do not Crash: Wifi connected before the loop");
  #endif

  // DO NOT CRASH with a single controller 
  FastLED.addLeds<WS2812B, DATA_PIN1>(leds, NUM_LEDS); 

  #ifdef FASTLED_SINGLE_CONTROLLER
    Serial.println("Do not Crash: FastLED runs a single controller");
  #else
    FastLED.addLeds<WS2812B, DATA_PIN2>(leds, NUM_LEDS); 
  #endif

  // DO NOT CRASH with FastLED < 3.4.0
  #if (FASTLED_VERSION < 3004000)
    Serial.println("Do not Crash: FastLED version < 3.4.0"); 
  #endif
}

void loop()
{
  FastLED.show();
  delay(1); // the smaller the delay, the more it crashes
}
```

### Versions
```
PLATFORM: Espressif 32 (2.1.0)
PACKAGES: framework-arduinoespressif32 3.10004.201016 (1.0.4)
FastLED:  3.4.0
WiFi:     1.0
```
 ### Stack dump
```
Guru Meditation Error: Core  1 panic'ed (Cache disabled but cached memory region accessed)
Core 1 register dump:
PC      : 0x4012737c  PS      : 0x00060034  A0      : 0x800810e5  A1      : 0x3ffbe740
A2      : 0x3ffc1160  A3      : 0x00000000  A4      : 0x800d1956  A5      : 0x3ffb1eb0
A6      : 0x00000002  A7      : 0x00000000  A8      : 0x80081091  A9      : 0x00000000
A10     : 0x00000000  A11     : 0x00000000  A12     : 0x0000000d  A13     : 0x3ffb1e90
A14     : 0x00000002  A15     : 0x00000000  SAR     : 0x00000020  EXCCAUSE: 0x00000007
EXCVADDR: 0x00000000  LBEG    : 0x00000000  LEND    : 0x00000000  LCOUNT  : 0x00000000
Core 1 was running in ISR context:
EPC1    : 0x40088cad  EPC2    : 0x00000000  EPC3    : 0x40086daa  EPC4    : 0x4012737c

Backtrace: 0x4012737c:0x3ffbe740 0x400810e2:0x3ffbe760 0x4008118a:0x3ffbe780 0x400811fa:0x3ffbe7b0 0x40084c39:0x3ffbe7d0 0x40086da7:0x3ffba190 0x40082f03:0x3ffba1b0 0x40088ef5:0x3ffba1d0
  #0  0x4012737c:0x3ffbe740 in rmt_set_pin at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/driver/rmt.c:393
  #1  0x400810e2:0x3ffbe760 in ESP32RMTController::startNext(int) at .pio\libdeps\lolin32\FastLED\src\platforms\esp\32/clockless_rmt_esp32.cpp:430 (discriminator 4)
  #2  0x4008118a:0x3ffbe780 in ESP32RMTController::doneOnChannel(rmt_channel_t, void*) at .pio\libdeps\lolin32\FastLED\src\platforms\esp\32/clockless_rmt_esp32.cpp:430 (discriminator 4)
  #3  0x400811fa:0x3ffbe7b0 in ESP32RMTController::interruptHandler(void*) at .pio\libdeps\lolin32\FastLED\src\platforms\esp\32/clockless_rmt_esp32.cpp:430 (discriminator 4)
  #4  0x40084c39:0x3ffbe7d0 in _xt_medint3 at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/freertos/xtensa_vectors.S:1323
  #5  0x40086da7:0x3ffba190 in spi_flash_op_block_func at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/spi_flash/cache_utils.c:82 (discriminator 1)
  #6  0x40082f03:0x3ffba1b0 in ipc_task at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/esp32/ipc.c:62
  #7  0x40088ef5:0x3ffba1d0 in vPortTaskWrapper at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/freertos/port.c:355 (discriminator 1)
```

**Comment by sebdelsol** [comment]
with `#define FASTLED_RMT_BUILTIN_DRIVER 1` it crashes way less often:

```
Guru Meditation Error: Core  1 panic'ed (Interrupt wdt timeout on CPU1)
Core 1 register dump:
PC      : 0x4008119e  PS      : 0x00050034  A0      : 0x40084c3c  A1      : 0x3ffbe7b0
A2      : 0x00000003  A3      : 0x00000001  A4      : 0x0000001a  A5      : 0x4008b658
A6      : 0x3ffc11c8  A7      : 0x00000000  A8      : 0x3ffc1098  A9      : 0x00000006
A10     : 0x3ffc11c8  A11     : 0x00000000  A12     : 0x800810c5  A13     : 0x3ffb1e60
A14     : 0x3ffc11c8  A15     : 0x00000001  SAR     : 0x00000014  EXCCAUSE: 0x00000006
EXCVADDR: 0x00000000  LBEG    : 0x4000c46c  LEND    : 0x4000c477  LCOUNT  : 0x00000000
Core 1 was running in ISR context:
EPC1    : 0x400810c5  EPC2    : 0x00000000  EPC3    : 0x400810e5  EPC4    : 0x4008119e

Backtrace: 0x4008119e:0x3ffbe7b0 0x40084c39:0x3ffbe7d0 0x400810e2:0x3ffb1ea0 0x4008126f:0x3ffb1ec0 0x400d198c:0x3ffb1ee0 0x400d16ff:0x3ffb1f00 0x400d1147:0x3ffb1f50 0x400d1d45:0x3ffb1f90 0x400d3e21:0x3ffb1fb0 0x40088ef5:0x3ffb1fd0
  #0  0x4008119e:0x3ffbe7b0 in ESP32RMTController::interruptHandler(void*) at .pio\libdeps\lolin32\FastLED\src\platforms\esp\32/clockless_rmt_esp32.cpp:83
  #1  0x40084c39:0x3ffbe7d0 in _xt_medint3 at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/freertos/xtensa_vectors.S:1323
  #2  0x400810e2:0x3ffb1ea0 in ESP32RMTController::startNext(int) at .pio\libdeps\lolin32\FastLED\src\platforms\esp\32/clockless_rmt_esp32.cpp:83
  #3  0x4008126f:0x3ffb1ec0 in ESP32RMTController::showPixels() at .pio\libdeps\lolin32\FastLED\src\platforms\esp\32/clockless_rmt_esp32.cpp:83
  #4  0x400d198c:0x3ffb1ee0 in ClocklessController<14, 60, 150, 90, (EOrder)10, 0, false, 5>::showPixels(PixelController<(EOrder)10, 1, 4294967295u>&) at .pio\libdeps\lolin32\FastLED\src/controller.h:171
  #5  0x400d16ff:0x3ffb1f00 in CPixelLEDController<(EOrder)10, 1, 4294967295u>::show(CRGB const*, int, CRGB) at .pio\libdeps\lolin32\FastLED\src/controller.h:171
  #6  0x400d1147:0x3ffb1f50 in CLEDController::showLeds(unsigned char) at .pio\libdeps\lolin32\FastLED\src/FastLED.cpp:269
      (inlined by) CFastLED::show(unsigned char) at .pio\libdeps\lolin32\FastLED\src/FastLED.cpp:59
  #7  0x400d1d45:0x3ffb1f90 in CFastLED::show() at .pio\libdeps\lolin32\FastLED\src/controller.h:171
      (inlined by) loop() at src/main.cpp:51
  #8  0x400d3e21:0x3ffb1fb0 in loopTask(void*) at C:\Users\c\.platformio\packages\framework-arduinoespressif32\cores\esp32/main.cpp:19
  #9  0x40088ef5:0x3ffb1fd0 in vPortTaskWrapper at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/freertos/port.c:355 (discriminator 1)

Core 0 register dump:
PC      : 0x4013bdde  PS      : 0x00060834  A0      : 0x800d6d6a  A1      : 0x3ffbbff0
A2      : 0x00000000  A3      : 0x00000001  A4      : 0x00000000  A5      : 0x00000001
A6      : 0x00060320  A7      : 0x00000000  A8      : 0x800d5db2  A9      : 0x3ffbbfc0
A10     : 0x00000000  A11     : 0x00000001  A12     : 0x80089108  A13     : 0x3ffb5570
A14     : 0x00000000  A15     : 0x3ffbbce0  SAR     : 0x00000000  EXCCAUSE: 0x00000006
EXCVADDR: 0x00000000  LBEG    : 0x00000000  LEND    : 0x00000000  LCOUNT  : 0x00000000

Backtrace: 0x4013bdde:0x3ffbbff0 0x400d6d67:0x3ffbc010 0x4008a6d9:0x3ffbc030 0x40088ef5:0x3ffbc050
  #0  0x4013bdde:0x3ffbbff0 in esp_pm_impl_waiti at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/esp32/pm_esp32.c:492
  #1  0x400d6d67:0x3ffbc010 in esp_vApplicationIdleHook at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/esp32/freertos_hooks.c:86
  #2  0x4008a6d9:0x3ffbc030 in prvIdleTask at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/freertos/tasks.c:3507
  #3  0x40088ef5:0x3ffbc050 in vPortTaskWrapper at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/freertos/port.c:355 (discriminator 1)
```

**Comment by sebdelsol** [comment]
#1156 seems to improve the issue but it stills crash

**Comment by sebdelsol** [comment]
#1156 suggestion code seems to help but it still crashes.

**Comment by sebdelsol** [comment]
When Wifi connects NVS data is written except when `WiFi.persistent(false);` is called.
So it seems to be the very same issue biologist79/Tonuino-ESP32-I2S#29.

#1144 alleviate the issue but doesn't completely fix it.

**Comment by sebdelsol** [comment]
Fixed by #1163
Thanks a lot
